Add prior weights function calculate_snp_weights() in MendelIHT_utilities.jl

gdmosher opened the pull request.

klkeys committed:
Merge pull request #6 from biona001/master

Doubly sparse projection for MendelIHT

gdmosher committed:
Commit weights work! MAF weights are selected

Add function calculatePriorWeightsforIHT()
Non-standard configuration
Requires IHT
Runs in Jupyter by using these commands
cd("F:")
cd("\\JuliaPro-0.6.3.1-MKL\\pkgs-0.6.3.1\\v0.6\\IHTBen\\test\\")
using IHT
pwd()
using SnpArrays
include("..\\src\\MendelIHT.jl")
MendelIHT2("gwas 1 Control.txt")
# Note: On a freshley restarted kernel, this call always crashes with the following "world age" error message, 
#            just run this cell again (on the same kernel) and it will work fine. Sorry no idea how to fix it. Gordon.
#MethodError: no method matching setcharheight(::Float64)
#The applicable method may be too new: running in world age 22726, while current world is 22727.

gdmosher committed:
Add function printConvergenceReport()

Had to return a bunch more variables from L0_reg() to support this feature.

gdmosher committed:
Add keywords for weighting, start cleanup of USE_INTERCEPT

still need to validate user input of keywords

gdmosher committed:
Finished cleanup of USE_INTERCEPT, commit snippets

gdmosher committed:
More cleanup for merge and get snpdata out of MendelIHT.jl

gdmosher committed:
Reverse merge is done, I have all of Ben's code from tonight

It runs good! Updated, but haven't tried the unit tests yet.

gdmosher committed:
Clean up for pull request

biona001 commented on src/MendelIHT.jl:
```diff
@@ -1,362 +1,339 @@
+    keyword["pw_pathway2_factorvalue"] = ""
+    keyword["pw_pathway3"] = ""
+    keyword["pw_pathway3_constantvalue"] = ""
+    keyword["pw_pathway3_factorvalue"] = ""
```
Is there really a need to define so many different keywords?

biona001 commented on src/calculatePriorWeightsforIHT.jl:
```diff
@@ -0,0 +1,135 @@
+)
+    # get my_snpMAF from x
+    ALLELE_MAX = 2 * size(x,1)
+    maf, minor_allele, missings_per_snp, missings_per_person = summarize(x)
```
`summarize()` is an expensive operation because it requires looping through the whole genotype matrix. Consider making `maf` an input of `calculatePriorWeights`.

Also consider changing the function name to something like `calculate_snp_weights` or something

biona001 reviewed (commented):
You need to also change the `README` file to instruct users on how to run `MendelIHT` with the extra weighting functionalities

gdmosher committed:
Fix line endings on 4 files

gdmosher committed:
function calculate_nsp_weights() is now in MendelIHT_utilities.jl

gdmosher committed:
dummy commit 6

gdmosher committed:
Reduce list of keywords to one: prior weights = maf

gdmosher committed:
Fix more line endings

gdmosher committed:
Merge branch 'Gordon2merge' into Gordon3

gdmosher committed:
Merge pull request #1 from gdmosher/Gordon3

Merge Ben's code

gdmosher committed:
Clean up merge typos

gdmosher committed:
It runs with "gwas 1 Control.txt" now

Pkg.test("IHT") doesn't work on my Windows system.
Uncomment "prior_weights = maf" in control file to apply maf weights.
